core/src/main/java/com/graphhopper/util/PathMerger.java

```diff
@@ -142,6 +142,8 @@ public void doWork(PathWrapper altRsp, List<Path> paths, Translation tr) {
         if (allFound && simplifyResponse && (calcPoints || enableInstructions)) {
             PathSimplification ps = new PathSimplification(altRsp, douglasPeucker, enableInstructions);
             ps.simplify();
+            // Make sure that the instruction references are not broken
+            altRsp.getPoints().makeImmutable();
```

Comment:
I wasn't sure if it would be better to put this in the simplification, but then decided it makes more sense here, so it's not hidden and it's more obvious to others what happens here.

Reply:
Good question :)

I would tend to put it in PathSimplication as then all uses of it are already safe. 

BTW: what happens if we have an immutable PointList and pass it to PathSimplification - will this fail?

Reply:
Ok, I can change this. 

> BTW: what happens if we have an immutable PointList and pass it to PathSimplification - will this fail?

Haven't tried it, but most probably, yes. Right now I haven't added a way to unlock a PointList. Once Locked it stays locked. You will have to create a deep copy to have the same PointList unlocked, which kind of makes sense to me, as we promised that the safeShallowCopy would be actually safe.